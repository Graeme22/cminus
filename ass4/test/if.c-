// C-F20
main()
{
    if (1>2) 3;
    if (21<22) 23; else 24;
    if (31==32) if (33==34) 35; else 36;
    if (41==42) if (43==44) 45; else 46;  else 47;

    if (51==52)
         if (53==54) 151;
         else 152;
    else
        if (56==57) 58;
        else 59;

    if (61==1) 11;
    else if (62==1) 12;
    else if (63==1) 13;
    else 14;
}




