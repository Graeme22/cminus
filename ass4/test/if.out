====================================
FILE: if.c-
Func main: returns type void [line: 2]
.   Child: 1  Compound [line: 3]
.   .   Child: 1  If [line: 4]
.   .   .   Child: 0  Op > : type bool [line: 4]
.   .   .   .   Child: 0  Const 1 : type int [line: 4]
.   .   .   .   Child: 1  Const 2 : type int [line: 4]
.   .   .   Child: 1  Const 3 : type int [line: 4]
.   .   Sibling: 1  If [line: 5]
.   .   .   Child: 0  Op < : type bool [line: 5]
.   .   .   .   Child: 0  Const 21 : type int [line: 5]
.   .   .   .   Child: 1  Const 22 : type int [line: 5]
.   .   .   Child: 1  Const 23 : type int [line: 5]
.   .   .   Child: 2  Const 24 : type int [line: 5]
.   .   Sibling: 2  If [line: 6]
.   .   .   Child: 0  Op == : type bool [line: 6]
.   .   .   .   Child: 0  Const 31 : type int [line: 6]
.   .   .   .   Child: 1  Const 32 : type int [line: 6]
.   .   .   Child: 1  If [line: 6]
.   .   .   .   Child: 0  Op == : type bool [line: 6]
.   .   .   .   .   Child: 0  Const 33 : type int [line: 6]
.   .   .   .   .   Child: 1  Const 34 : type int [line: 6]
.   .   .   .   Child: 1  Const 35 : type int [line: 6]
.   .   .   .   Child: 2  Const 36 : type int [line: 6]
.   .   Sibling: 3  If [line: 7]
.   .   .   Child: 0  Op == : type bool [line: 7]
.   .   .   .   Child: 0  Const 41 : type int [line: 7]
.   .   .   .   Child: 1  Const 42 : type int [line: 7]
.   .   .   Child: 1  If [line: 7]
.   .   .   .   Child: 0  Op == : type bool [line: 7]
.   .   .   .   .   Child: 0  Const 43 : type int [line: 7]
.   .   .   .   .   Child: 1  Const 44 : type int [line: 7]
.   .   .   .   Child: 1  Const 45 : type int [line: 7]
.   .   .   .   Child: 2  Const 46 : type int [line: 7]
.   .   .   Child: 2  Const 47 : type int [line: 7]
.   .   Sibling: 4  If [line: 9]
.   .   .   Child: 0  Op == : type bool [line: 9]
.   .   .   .   Child: 0  Const 51 : type int [line: 9]
.   .   .   .   Child: 1  Const 52 : type int [line: 9]
.   .   .   Child: 1  If [line: 10]
.   .   .   .   Child: 0  Op == : type bool [line: 10]
.   .   .   .   .   Child: 0  Const 53 : type int [line: 10]
.   .   .   .   .   Child: 1  Const 54 : type int [line: 10]
.   .   .   .   Child: 1  Const 151 : type int [line: 10]
.   .   .   .   Child: 2  Const 152 : type int [line: 11]
.   .   .   Child: 2  If [line: 13]
.   .   .   .   Child: 0  Op == : type bool [line: 13]
.   .   .   .   .   Child: 0  Const 56 : type int [line: 13]
.   .   .   .   .   Child: 1  Const 57 : type int [line: 13]
.   .   .   .   Child: 1  Const 58 : type int [line: 13]
.   .   .   .   Child: 2  Const 59 : type int [line: 14]
.   .   Sibling: 5  If [line: 16]
.   .   .   Child: 0  Op == : type bool [line: 16]
.   .   .   .   Child: 0  Const 61 : type int [line: 16]
.   .   .   .   Child: 1  Const 1 : type int [line: 16]
.   .   .   Child: 1  Const 11 : type int [line: 16]
.   .   .   Child: 2  If [line: 17]
.   .   .   .   Child: 0  Op == : type bool [line: 17]
.   .   .   .   .   Child: 0  Const 62 : type int [line: 17]
.   .   .   .   .   Child: 1  Const 1 : type int [line: 17]
.   .   .   .   Child: 1  Const 12 : type int [line: 17]
.   .   .   .   Child: 2  If [line: 18]
.   .   .   .   .   Child: 0  Op == : type bool [line: 18]
.   .   .   .   .   .   Child: 0  Const 63 : type int [line: 18]
.   .   .   .   .   .   Child: 1  Const 1 : type int [line: 18]
.   .   .   .   .   Child: 1  Const 13 : type int [line: 18]
.   .   .   .   .   Child: 2  Const 14 : type int [line: 19]
Number of warnings: 0
Number of errors: 0
